packages/editor-ui/src/modules/ui.ts

```diff
@@ -0,0 +1,72 @@
+import { RENAME_MODAL_KEY, SAVE_AS_MODAL_KEY, TAGS_MANAGER_MODAL_KEY, WORKLOW_OPEN_MODAL_KEY } from '@/constants';
+import Vue from 'vue';
+import { ActionContext, Module } from 'vuex';
+import {
+	IRootState,
+	IUiState,
+} from '../Interface';
+
+const module: Module<IUiState, IRootState> = {
+	namespaced: true,
+	state: {
+		modals: {
+			[SAVE_AS_MODAL_KEY]: {
+				open: false,
+			},
+			[RENAME_MODAL_KEY]: {
+				open: false,
+			},
+			[TAGS_MANAGER_MODAL_KEY]: {
+				open: false,
+			},
+			[WORKLOW_OPEN_MODAL_KEY]: {
+				open: false,
+			},
+		},
+		modalStack: [],
+		sidebarMenuCollapsed: true,
+	},
+	getters: {
+		isModalOpen: (state: IUiState) => {
+			return (name: string) => state.modals[name].open;
+		},
+		isModalActive: (state: IUiState) => {
+			return (name: string) => name === state.modalStack[0];
+		},
+		anyModalsOpen: (state: IUiState) => {
+			return state.modalStack.length > 0;
+		},
+		sidebarMenuCollapsed: (state: IUiState): boolean => state.sidebarMenuCollapsed,
+	},
+	mutations: {
+		openModal: (state: IUiState, name: string) => {
+			Vue.set(state.modals[name], 'open', true);
+			state.modalStack = [name].concat(state.modalStack);
+		},
+		closeTopModal: (state: IUiState) => {
+			const name = state.modalStack[0];
+			Vue.set(state.modals[name], 'open', false);
+
+			state.modalStack = state.modalStack.slice(1);
+		},
```

Comment:
because I like `state.modalStack[0]` more than `stack.modalStack[state.modalStack.length - 1]`. it's just a preference. I can change it if you think pop/push is clearer

Reply:
oh I don't think we can use push/pop because of vue reactivity rules https://vuejs.org/v2/guide/reactivity.html#For-Arrays

Reply:
Thank you. Not used to seeing a stack with reassignment instead of mutation.